datajunction-server/datajunction_server/models/sql.py

```diff
@@ -0,0 +1,55 @@
+"""
+Models for generated SQL
+"""
+from typing import List, Optional
+
+from pydantic.class_validators import root_validator
+from pydantic.main import BaseModel
+
+from datajunction_server.models.engine import Dialect
+from datajunction_server.models.query import ColumnMetadata
+from datajunction_server.transpilation import get_transpilation_plugin
+from datajunction_server.utils import get_settings
+
+
+class NodeNameVersion(BaseModel):
+    """
+    Node name and version
+    """
+
+    name: str
+    version: str
+
+    class Config:  # pylint: disable=missing-class-docstring,too-few-public-methods
+        orm_mode = True
+
+
+class GeneratedSQL(BaseModel):
+    """
+    Generated SQL for a given node, the output of a QueryBuilder(...).build() call.
+    """
+
+    node: NodeNameVersion
+    sql: str
+    columns: Optional[List[ColumnMetadata]] = None  # pragma: no-cover
+    dialect: Optional[Dialect] = None
+    upstream_tables: Optional[List[str]] = None
+
+    @root_validator(pre=False)
+    def transpile_sql(  # pylint: disable=no-self-argument
+        cls,
+        values,
+    ):
+        """
+        Transpiles SQL to the specified dialect with the configured transpilation plugin.
+        If no plugin is configured, it will just return the original generated query.
+        """
+        settings = get_settings()
```

Comment:
@samredai recently mentioned that we should be depending on this function rather than importing it directly, but I may be missing some context here.

Reply:
Yeah that's right, I realized this when I was working on some stuff in the query service. Anywhere we just call it like this without using `Depends` won't get replaced when we try to replace it with dependency injection. I'm not sure how deep the call stack is to get here but it might be better to just make `sql_transpilation_library` an argument to the function and then pass it in from the endpoint function that uses `settings: Settings = Depends(get_settings)`.

Reply:
Hmm the issue is that this is a pydantic model's `root_validator`, so we can't actually add any extra arguments here, since it's pydantic that is making the call to this method. I think what we can do instead is add `sql_transpilation_library ` to the pydantic model itself and then use that to modify the object in the `root_validator`.